stubs/tensorflow/tensorflow/data/__init__.pyi

```diff
@@ -223,8 +223,17 @@ class Dataset(ABC, Generic[_T1]):
         name: str | None = None,
     ) -> Dataset[Dataset[_T1]]: ...
     def with_options(self, options: Options, name: str | None = None) -> Dataset[_T1]: ...
+    @overload
+    @staticmethod
+    def zip(*args: Collection[Dataset[Any]], name: str | None = None) -> Dataset[tuple[Any, ...]]: ...
```

Comment:
In addition to collection I think it should be ContainerGeneric[Dataset[Any]] here. The documentation allows arbitrarily nested types holding dataset. The return type though can’t be described at that point beyond Any.

Reply:
Done in a7f33d175